mobile/src/main/java/ch/epfl/sdp/mobile/infrastructure/persistence/auth/Auth.kt

```diff
@@ -21,6 +21,24 @@ interface Auth {
     /** Indicates that there was a failure during authentication. */
     object FailureInternal : AuthenticationResult
 
+    /** Indicates that the password the user chose for their account is not strong enough. */
+    object FailureBadPassword : AuthenticationResult
+
+    /** Indicates that the email the user typed is malformed. */
+    object FailureIncorrectEmailFormat : AuthenticationResult
+
+    /** Indicates that an account already exists with the given email. */
+    object FailureExistingAccount : AuthenticationResult
+
+    /** Indicates that the given password associated to an email is incorrect. */
+    object FailureIncorrectPassword : AuthenticationResult
+
+    /**
+     * Indicates that the user account corresponding to an email does not exist or has been
+     * disabled.
+     */
+    object FailureInvalidUser : AuthenticationResult
```

Comment:
Question : why not have a hierarchy that makes sure all failures are of the same type ? Something along the lines of :

```kotlin
sealed interface AuthenticationResult {
  data class Success(val user: User?) : AuthenticationResult
  sealed interface Failure : AuthenticationResult {
    object Internal : Failure
    object BadPassword : Failure
  }
}
```

Reply:
This makes checking for failure much easier :

```
val result: AuthenticationResult = // TODO ...
if (result is AuthenticationResult.Failure) {
  println("Something went wrong")
}

Reply:
Fixed in 12d715c